internal/support/cronscheduler/application/scheduleactionrecord.go

```diff
@@ -109,10 +115,136 @@ func DeleteScheduleActionRecordsByAge(ctx context.Context, age int64, dic *di.Co
 	return nil
 }
 
+// GenerateMissedScheduleActionRecords generates missed schedule action records
+func GenerateMissedScheduleActionRecords(ctx context.Context, dic *di.Container, job models.ScheduleJob, latestRecords []models.ScheduleActionRecord) errors.EdgeX {
+	dbClient := container.DBClientFrom(dic.Get)
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	correlationId := correlation.FromContext(ctx)
+
+	// All latest records should have the same creation time
+	lastRecordTimestamp := latestRecords[0].Created
```

Comment:
don't access the harcode index dirctly
if the slice is empty, it will cause the service panic

use `ScheduledAt` instead of `Created`

Reply:
Use for loop to check each record in the next PR (when action uuid is ready) to prevent the missing record happened in the middle of the schedule run.
For example, the scheduld is triggered with 100 actions, and the system crashed in the 70th action running.